solutions/distinct_pair_sum.py

```diff
@@ -0,0 +1,65 @@
+def distinct_pair_sum(list, k):
+
+    pairs = {}
+
+    for i in range(len(list) - 1):
+        if list[i] + list[i + 1] == k:
+            if list[i] not in pairs and list[i + 1] not in pairs:
+                pairs[list[i]] = [list[i], list[i + 1]]
+
+    return [pairs[key] for key in pairs]
```

Comment:
Python dictionaries have the `values` method to retrieve all the values in a dictionary. 

Note that the `values` method will return a `dict_values` object so it needs to be converted using `list(pairs.values())`